components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -133,6 +133,9 @@ message Workspace {
 
     // status is the current status of this Workspace.
     WorkspaceStatus status = 6;
+
+    // recent_folders is the opened folders inside the workspace. Used to determine the folder path to load the workspace in.
+    repeated string recent_folders = 7;
```

Comment:
How many entries will be in this list? What if it grows very large?

Will a workspace, across instances, always have the same set of `recet_folders`? If not, then it would make sense to push it into Status, maybe?

Reply:
> How many entries will be in this list? What if it grows very large?

It will be fine to put a hard cap server side, as it's name suggest it's just recent entries, if the cap if reached then old entries should be discarded.

> Will a workspace, across instances, always have the same set of recet_folders? If not, then it would make sense to push it into Status, maybe?

Yeah not the case, in the near future I expect to have an api to update this entry [like in codespaces](https://docs.github.com/en/rest/codespaces/codespaces?apiVersion=2022-11-28#update-a-codespace-for-the-authenticated-user), so I guess it depends on how that `updateWorkspace` api is implemented, will it require a running instance? if not how would that work? update the `recet_folders` in the stopped instance status? I think it makes more sense to be global as there's only one instance at a time right?

Reply:
> I'm inclined to push this into Status, as it better reflects the expectation that it's read only, and it's something that resulted from the specification, not that the user chose it.

Does this mean it cannot be updated in the current running instance? it should, I can open any folder and be able to update the list

Reply:
Yes, it can be updated in the running instance, or terminated instance. Basically can be updated any time, but only by our systems, never by the user.